web/src/beta/lib/core/engines/Cesium/utils.ts

```diff
@@ -151,6 +189,63 @@ export function findEntity(
   return;
 }
 
+export function findFeaturesFromLayer<T = Entity | Cesium3DTileset | InternalCesium3DTileFeature>(
```

Comment:
If possible, It'd be great if you can breakdown this function into smaller[OPTIONAL].

function filterEntitiesById(entities: Entity[], layerId: string, featureId: string[]): Entity[] {
  // filtering logic
}

function getEntitiesFromDatasources(viewer: CesiumViewer, layerId: string, featureId: string[]): Entity[] {
  // logic to get entities from datasources
}

function get3DTileFeatures(viewer: CesiumViewer, featureId: string[]): Set<Entity | Cesium3DTileset | InternalCesium3DTileFeature> {
  // logic to get entities from 3D Tiles
}